shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs

```diff
@@ -240,68 +250,74 @@ text64FromCBOR = do
   fromCBOR = Url <$> text64FromCBOR
 
-newtype DnsName = DnsName { dnsToText :: Text }
+newtype DnsName = DnsName {dnsToText :: Text}
   deriving (Eq, Generic, Show, ToCBOR, NoUnexpectedThunks)
 
 textToDns :: Text -> Maybe DnsName
 textToDns t = DnsName <$> text64 t
 
-instance FromCBOR DnsName
- where
+instance FromCBOR DnsName where
   fromCBOR = DnsName <$> text64FromCBOR
 
-newtype Port = Port { portToWord16 :: Word16 }
+newtype Port = Port {portToWord16 :: Word16}
   deriving (Eq, Ord, Num, Generic, Show, ToCBOR, FromCBOR, NoUnexpectedThunks)
 
 --------------------------------------------------------------------------------
 -- Active Slot Coefficent, named f in
 -- "Ouroboros Praos: An adaptively-secure, semi-synchronous proof-of-stake protocol"
 --------------------------------------------------------------------------------
 
-data ActiveSlotCoeff =
-  ActiveSlotCoeff
-  { unActiveSlotVal :: !UnitInterval
-  , unActiveSlotLog :: !Integer  -- TODO mgudemann make this FixedPoint,
-                                 -- currently a problem because of
-                                 -- NoUnexpectedThunks instance for FixedPoint
-  } deriving (Eq, Ord, Show, Generic)
+data ActiveSlotCoeff = ActiveSlotCoeff
+  { unActiveSlotVal :: !UnitInterval,
+    unActiveSlotLog :: !Integer -- TODO mgudemann make this FixedPoint,
+      -- currently a problem because of
+      -- NoUnexpectedThunks instance for FixedPoint
+  }
+  deriving (Eq, Ord, Show, Generic)
 
 instance NoUnexpectedThunks ActiveSlotCoeff
 
-instance FromCBOR ActiveSlotCoeff
- where
-   fromCBOR = do
-     v <- fromCBOR
-     pure $ mkActiveSlotCoeff v
-
-instance ToCBOR ActiveSlotCoeff
- where
-   toCBOR (ActiveSlotCoeff { unActiveSlotVal = slotVal
-                           , unActiveSlotLog = _logVal}) =
-     toCBOR slotVal
+instance FromCBOR ActiveSlotCoeff where
+  fromCBOR = do
+    v <- fromCBOR
+    pure $ mkActiveSlotCoeff v
+
+instance ToCBOR ActiveSlotCoeff where
+  toCBOR
+    ( ActiveSlotCoeff
+        { unActiveSlotVal = slotVal,
+          unActiveSlotLog = _logVal
+        }
+      ) =
+      toCBOR slotVal
 
 mkActiveSlotCoeff :: UnitInterval -> ActiveSlotCoeff
 mkActiveSlotCoeff v =
-  ActiveSlotCoeff { unActiveSlotVal = v
-                  , unActiveSlotLog =
-                    if (intervalValue v) == 1
-                      -- If the active slot coefficient is equal to one,
-                      -- then nearly every stake pool can produce a block every slot.
-                      -- In this degenerate case, where ln (1-f) is not defined,
-                      -- we set the unActiveSlotLog to zero.
-                      then 0
-                      else floor (fpPrecision * (
-                        ln' $ (1 :: FixedPoint) - (fromRational $ intervalValue v))) }
+  ActiveSlotCoeff
+    { unActiveSlotVal = v,
+      unActiveSlotLog =
+        if (intervalValue v) == 1
+          then-- If the active slot coefficient is equal to one,
```

Comment:
Random drive-by commentary: `then--` is weird

Reply:
Yep, I mentioned this to Mark, he said it's a known bug that will be fixed as part of the current project :-)